services/wallet/activity/activity.go

```diff
@@ -174,6 +195,42 @@ func (e *Entry) PayloadType() PayloadType {
 	return e.payloadType
 }
 
+func (e *Entry) isNFT() bool {
+	tt := e.transferType
+	return tt != nil && (*tt == TransferTypeErc721 || *tt == TransferTypeErc1155) && ((e.tokenIn != nil && e.tokenIn.TokenID != nil) || (e.tokenOut != nil && e.tokenOut.TokenID != nil))
+}
+
+// TODO - #11952: use only one of (big.Int, bigint.BigInt and hexutil.Big)
+func tokenIDToWalletBigInt(tokenID *hexutil.Big) *bigint.BigInt {
+	if tokenID == nil {
+		return nil
+	}
+
+	bi := new(big.Int).Set((*big.Int)(tokenID))
+	return &bigint.BigInt{Int: bi}
+}
+
+func (e *Entry) anyIdentity() *thirdparty.CollectibleUniqueID {
+	if e.tokenIn != nil {
+		return &thirdparty.CollectibleUniqueID{
+			ContractID: thirdparty.ContractID{
+				ChainID: e.tokenIn.ChainID,
+				Address: e.tokenIn.Address,
+			},
+			TokenID: tokenIDToWalletBigInt(e.tokenIn.TokenID),
+		}
+	} else if e.tokenOut != nil {
+		return &thirdparty.CollectibleUniqueID{
+			ContractID: thirdparty.ContractID{
+				ChainID: e.tokenOut.ChainID,
+				Address: e.tokenOut.Address,
+			},
+			TokenID: tokenIDToWalletBigInt(e.tokenIn.TokenID),
```

Comment:
shouldn't this be `tokenOut`?

Reply:
🤦🏻 Thanks, @dlipicar for the eagle eyes 👀